tools/clang/test/CodeGenHLSL/quick-test/inline_dbginfo.hlsl

```diff
@@ -0,0 +1,78 @@
+// RUN: %dxc -T ps_6_0 -E main -Zi %s | FileCheck %s
+
+// Make sure llvm.dbg.value exist.
+// CHECK: call void @llvm.dbg.value
+
+
+typedef float4 MyCoolFloat4;
+
+static float4 myStaticGlobalVar = float4(1.0, 1.0, 1.0, 1.0);
+
+
+// Local var with same name as outer scope
```

Comment:
why is the name identifier suspected to have an impact on this?

Reply:
It is not my test, my guess is
localScopeVar_func has local variable "color", main has parameter "color".
Debug info should be able to know there're 2 different "color".